gbpservice/nfp/orchestrator/modules/service_orchestrator.py

```diff
@@ -778,25 +782,79 @@ def handle_update_user_config(self, event):
             request_data['network_function_id'])
         stack_id = network_function_details['network_function'
                                             ]['heat_stack_id']
-        # Heat driver to update stack
-        request_data['heat_stack_id'] = self.config_driver.update_config(
-            network_function_details, stack_id)
         network_function = network_function_details['network_function']
-        request_data['network_function_id'] = network_function['id']
+        service_profile_id = network_function['service_profile_id']
+        service_type = self._get_service_type(service_profile_id)
+        if service_type == pconst.VPN or service_type == pconst.FIREWALL:
+            service_chain_id = network_function['service_chain_id']
+            admin_token = self.keystoneclient.get_admin_token()
+            servicechain_instance = self.gbpclient.get_servicechain_instance(
+                            admin_token,
+                            service_chain_id)
+            provider_ptg_id = servicechain_instance['provider_ptg_id']
+            provider_ptg = self.gbpclient.get_policy_target_group(
+                            admin_token,
+                            provider_ptg_id)
+            provider_tenant_id = provider_ptg['tenant_id']
+            stack_id = self.config_driver.delete_config(stack_id,
+                                                        provider_tenant_id)
+            request_data = {
+                    'heat_stack_id': stack_id,
+                    'network_function_id': network_function['id'],
+                    'tenant_id': provider_tenant_id,
+                    'action': 'update',
+                    'operation': request_data['operation']
+            }
+            self._create_event('DELETE_USER_CONFIG_IN_PROGRESS',
+                               event_data=request_data,
+                               is_poll_event=True, original_event=event)
+        else:
+            self._create_event('CONTINUE_UPDATE_USER_CONFIG',
+                               event_data=event.data,
+                               is_internal_event=True)
 
-        if not request_data['heat_stack_id']:
-            self._create_event('UPDATE_USER_CONFIG_FAILED',
-                               event_data=request_data)
+    def handle_continue_update_user_config(self, event):
+        request_data = event.data
+        network_function_details = self.get_network_function_details(
+            request_data['network_function_id'])
+        network_function = network_function_details['network_function']
+
+        if request_data['operation'] == 'update':
+            config_id = self.config_driver.update_config(
+                network_function_details,
+                network_function_details['network_function']['heat_stack_id'])
+        elif request_data['operation'] == 'consumer_add':
+            config_id = self.config_driver.handle_consumer_ptg_operations(
```

Comment:
why we get consumer_add, consumer_remove operations here, we already checked for that before, when we are creating event.